SMP: Fix keyboard navigation for sort dropdown

p-jackson opened the pull request:
#### Proposed Changes

Combo boxes/dropdown menus should be navigable using the keyboard. This PR fixes that accessibility issue.

This adds keyboard control using similar code to how Gutenberg's `<DropdownMenu>` component does it: 
https://github.com/WordPress/gutenberg/blob/b333d1d515c7fe91ec063112a6c178d5ed66b1e4/packages/components/src/dropdown-menu/index.js#L45
 
![Jan-31-2023 15-16-55](https://user-images.githubusercontent.com/1500769/215643062-1a459405-e138-467b-a97e-aff28feb8043.gif)



#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Test at `/sites`
* Sorting should work just like it used to
* Keyboard control should work on all supported browsers

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] ~[Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?~
- [ ] ~Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?~
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] ~Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)~
- [ ] ~Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?~
- [ ] ~For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?~
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

p-jackson committed:
Fix keyboard navigation for SMP sorting control

mpkelly reviewed (approved):
Tested and working with cursors in Chrome and Firefox. 

One small issue: the Status dropdown does not loop (e.g. jump from last to first item) whereas the Sort dropdown does. Maybe this could be changed in this PR too?

p-jackson commented:
Looks like `<NavigableMenu>` has a `cycle` prop which can be used to disable the cycling behaviour. To make it consistent I think it's easier to just disable cycling for the sort dropdown.

p-jackson committed:
Disable item cycling in sort menu

The status filter dropdown uses a calypso component which doesn't cycle
through options when you keep clicking "down". To make it consistent
it's easier to just disable cycling on the sort dropdown.

mpkelly commented:
I retested, and it's looking better - now both dropdown controls work the same way.
